### DISCUSSION TITLE
UI:  How do you move songs in the queue with iOS?

### DISCUSSION THREAD

**Discussion by fts-tmassey** [jcorporation/myMPD#345] [Q&A] [answered]
I cannot figure out how to move songs around in the queue on iOS.  On a desktop, I can use the mouse to click and drag.  Click and drag is always awkward on a finger-driven interface...  When try to "click" (place a finger down) and "drag" (move my finger up or down), the entire *list* scrolls up and down.  I've tried 3D push (i.e. push harder!) and touch-and-dwell then move, both with no effect.  No matter what, it just scrolls the song list.

Sorry to abuse the issue list here, but can someone give me a pointer on how to re-arrange songs in the queue on an iOS device?  Thank you very much!

Oh, and while I'm wasting your time:  a me-too on the "replace pagination with endless scrolling":  it makes finding the current song *very* difficult if you put your entire song library into the queue and use the shuffle queue feature (which I often do)...  And while I'm +1'ing, I would also suggest making the currently playing song MUCH MORE visible in the list.  Even when it *is* on the right page, finding the current song is still quite difficult;  the fact that it might not even *be* on the page I'm looking at is just an additional fun sharp edge to cut myself on at the same time...  :)  Dare I suggest a "go to current playing" or such option in the hamburger?

Tiny complaint aside, I've really enjoyed using myMPD.  More importantly, my non-techie wife is now using it happily too:  this is the first MPD client I've found that she has used not under duress...  Plus it's *way* ahead on features from any other MPD client I've used:  album art works, playlists work, random list shuffle works, it doesn't randomly dump my queue if I move too fast, etc.  All things I've had issues with using other iOS clients...  Bravo!  And thank you!

**Comment by jcorporation** [comment, marked as answer]
Thanks for your feedback.

Drag and Drop:
myMPD uses the native html5 drag and drop api to sort the queue. Unfortunately safari on iPhone does not support the api.  Currently there is no alternative way implemented.

Queue:
In devel there is a goto playing track button. It jumps to the page with the playing song.

Btw: why do you add your whole library to the queue and shuffle it? The jukebox mode has the same effect without a large queue.

I marked this issue as a question. If you want a feature please open a specific  feature request.

**Comment by fts-tmassey** [comment]
Drag and drop:  thanks for the pointer.  It seems to work on iPad, but not iPhone.  Grrr...  But obviously not an MyMPD issue.

I didn’t see a “question” option: now that I know, I’ll look harder next time.  As for Jukebox: I had no awareness of it.  A look around the UI certainly gives you zero hints...  I had to Google for it to find it (buried under MyMPD/Settings/Playback).  I’ll play with it.

Thank you very much for your help!  I appreciate it.
